packages/blockchain-wallet-v4-frontend/src/scenes/plugin/Activity/TransactionDetails/TransactionDetails.tsx

```diff
@@ -0,0 +1,289 @@
+          <FormattedMessage
+            id='plugin.activity.transactionDetails.transaction'
+            defaultMessage='Transaction'
+          />
+        </Text>
+      </Padding>
+
+      <Flex flexDirection='column' gap={16}>
+        <TransactionDetailsItem
+          label={
+            <FormattedMessage
+              id='plugin.activity.transactionDetails.nonce'
+              defaultMessage='Nonce'
+            />
+          }
+          value={
+            <Text size='14px' lineHeight='21px' color='white' weight={500}>
+              {transaction.nonce}
+            </Text>
+          }
+        />
+        <TransactionDetailsItem
+          label={
+            <FormattedMessage
+              id='plugin.activity.transactionDetails.amount'
+              defaultMessage='Amount'
+            />
+          }
+          value={
+            <CoinDisplay size='14px' lineHeight='21px' weight={500} color='white' coin={coin}>
+              {transaction.amount}
+            </CoinDisplay>
+          }
+        />
+        <TransactionDetailsItem
+          label={
+            <FormattedMessage
+              id='plugin.activity.transactionDetails.gasFee'
+              defaultMessage='Gas Fee'
+            />
+          }
+          value={
+            <CoinDisplay size='14px' lineHeight='21px' weight={500} color='white' coin={coin}>
+              {convertGweiToWei(transaction.gasPrice)}
+            </CoinDisplay>
+          }
+        />
+        <TransactionDetailsItem
+          label={
+            <FormattedMessage
+              id='plugin.activity.transactionDetails.gasFee'
+              defaultMessage='Gas Limit'
+            />
+          }
+          value={
+            <CoinDisplay size='14px' lineHeight='21px' weight={500} color='white' coin={coin}>
+              {convertGweiToWei(transaction.gasLimit)}
+            </CoinDisplay>
+          }
+        />
+        <TransactionDetailsItem
+          label={
+            <FormattedMessage
+              id='plugin.activity.transactionDetails.totalGasFee'
+              defaultMessage='Total Gas Fee'
+            />
+          }
+          value={
+            <CoinDisplay size='14px' lineHeight='21px' weight={500} color='white' coin={coin}>
+              {new BigNumber(transaction.gasPrice)
+                .multipliedBy(new BigNumber(transaction.gasLimit))
+                .toString()}
+            </CoinDisplay>
+          }
+        />
+        <TransactionDetailsItem
+          label={
+            <FormattedMessage
+              id='plugin.activity.transactionDetails.total'
+              defaultMessage='Total'
+            />
+          }
+          value={
+            <CoinDisplay size='14px' lineHeight='21px' weight={500} color='white' coin={coin}>
+              {new BigNumber(transaction.gasPrice)
+                .multipliedBy(new BigNumber(transaction.gasLimit))
+                .plus(transaction.amount)
+                .toString()}
```

Comment:
should we define some variable with `useMemo` for this kind of calculations? Same comment for all the calculations in this component. At least it will make code cleaner.